modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/buffer/IndexerRequestBufferExecutorWatcher.java

```diff
@@ -64,44 +62,23 @@ public IndexerRequestBufferExecutor getIndexerRequestBufferExecutor() {
 
 	@Activate
 	@Modified
-	protected void activate(Map<String, Object> properties) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_indexerRegistryConfiguration = ConfigurableUtil.createConfigurable(
 			IndexerRegistryConfiguration.class, properties);
-	}
-
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY
-	)
-	protected void addIndexerRequestBufferExecutor(
-		IndexerRequestBufferExecutor indexerRequestBufferExecutor,
-		Map<String, Object> properties) {
-
-		_indexerRequestBufferExecutors.put(
-			_getBufferedExecutionMode(properties),
-			indexerRequestBufferExecutor);
-	}
 
-	protected void removeIndexerRequestBufferExecutor(
-		IndexerRequestBufferExecutor indexerRequestBufferExecutor,
-		Map<String, Object> properties) {
-
-		_indexerRequestBufferExecutors.remove(
-			_getBufferedExecutionMode(properties));
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			bundleContext, IndexerRequestBufferExecutor.class, null,
+			(indexerRequestBufferExecutor, emitter) -> emitter.emit(
```

Comment:
@W2jx I just noticed this.. Here, what's the type of the `indexerRequestBufferExecutor` parameter?

I think you are implementation `ServiceReferencemapper` as a lambda here. The first parameter of `ServiceReferenceMapper.map` should be a `ServiceReference<S>`, here it's `ServiceReference<IndexerRequestBufferExecutor>`, so you shouldn't name this parameter as `indexerRequestBufferExecutor`; instead, you should name it `serviceReference`.

Please double check all your other pull requests and the pull requests already sent/merged, thanks!

cc @JanisZhang @slnn

Reply:
I will fix it.